woodwork/utils.py

```diff
@@ -27,28 +27,38 @@ def camel_to_snake(s):
     return re.sub('([a-z0-9])([A-Z])', r'\1_\2', s).lower()
 
 
-def _convert_input_to_set(semantic_tags, error_language='semantic_tags'):
+def _convert_input_to_set(semantic_tags, error_language='semantic_tags', validate=True):
     """Takes input as a single string, a list of strings, or a set of strings
         and returns a set with the supplied values. If no values are supplied,
         an empty set will be returned."""
     if not semantic_tags:
         return set()
 
-    if type(semantic_tags) not in [list, set, str]:
-        raise TypeError(f"{error_language} must be a string, set or list")
+    if validate:
+        _validate_tags_input(semantic_tags, error_language)
 
     if isinstance(semantic_tags, str):
         return {semantic_tags}
 
     if isinstance(semantic_tags, list):
         semantic_tags = set(semantic_tags)
 
-    if not all([isinstance(tag, str) for tag in semantic_tags]):
-        raise TypeError(f"{error_language} must contain only strings")
+    if validate:
+        _validate_string_tags(semantic_tags, error_language)
 
     return semantic_tags
 
 
+def _validate_tags_input(semantic_tags, error_language):
```

Comment:
Maybe change this to `_validate_tags_input_type` to be a little more clear what we are checking here?

Reply:
changed!